packages/server/lib/remote_states.ts

```diff
@@ -88,14 +79,14 @@ export class RemoteStates {
 
     // reset the remoteStates and originStack to the primary
     this.remoteStates = new Map(stateArray[0] ? [stateArray[0]] : [])
-    this.originStack = stateArray[0] ? [stateArray[0][0]] : []
+    this.currentOriginKey = this.primaryOriginKey
   }
 
   current (): Cypress.RemoteState {
-    return this.get(this.originStack[this.originStack.length - 1]) as Cypress.RemoteState
+    return this.get(this.currentOriginKey) as Cypress.RemoteState
   }
 
-  set (urlOrState: string | Cypress.RemoteState, options: { auth?: {}, isCrossOrigin?: boolean } = {}): Cypress.RemoteState {
+  set (urlOrState: string | Cypress.RemoteState, options: { auth?: {} } = {}, isPrimaryOrigin: boolean = true): Cypress.RemoteState {
```

Comment:
guessing this sets the default remote state when no `baseUrl` is set?

Reply:
logic is mostly here: https://github.com/cypress-io/cypress/blob/matth/cy-origin-async-attach/packages/server/lib/server-e2e.ts#L304

we set primary if we haven't visited anywhere yet, and we are not visiting from inside a spec bridge.